### PR TITLE
Reject arg == null in Array#sort

### PR DESCRIPTION
**Pull request by MaxGraey**
Reject this very unlikely case in runtime:
```ts
type Comparator = (a: i32, b: i32) => i32;
const arr = <i32[]>[1, 2, 3];
var cmp: Comparator = <Comparator>null;
arr.sort(cmp);
```

### CODE REVIEW

**Comment by dcodeIO** [comment]
This is currently possible because there is no `null` checking in flows yet. Essentially, flows would have to keep track of what's nullable and what's not (i.e. `if (somethingNullable) { /* here, somethingNullable is not nullable anymore */ }`, and emit an error when doing an assignment like in this case. Once that's there, the only way to break something like in your example would be to `changetype`, but the plan is that using `changetype` outside of the standard library will yield an error eventually, unless a flag like `--allowUnsafe` is given.

File: std/assembly/array.ts [hunk 1]
```diff
@@ -302,6 +302,9 @@ export class Array<T> {
   }
 
   sort(comparator: (a: T, b: T) => i32 = defaultComparator<T>()): this {
+    // TODO remove this when flow will allow trackcing null
+    assert(comparator, "The comparison function must be a function");
```
Comment (dcodeIO) [inline std/assembly/array.ts]: Let's make this an `assert(comparator); // The comparison function must be a function`

Comment (MaxGraey) [reply std/assembly/array.ts]: ok

Comment (MaxGraey) [reply std/assembly/array.ts]: Done!

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**